tests/receive_message_twice.go

```diff
@@ -0,0 +1,106 @@
+	Expect(err).Should(BeNil())
+
+	//
+	// Send a transaction to Subnet A to issue a Warp Message from the Teleporter contract to Subnet B
+	//
+	ctx := context.Background()
+
+	sendCrossChainMessageInput := teleportermessenger.TeleporterMessageInput{
+		DestinationChainID: subnetBInfo.BlockchainID,
+		DestinationAddress: fundedAddress,
+		FeeInfo: teleportermessenger.TeleporterFeeInfo{
+			ContractAddress: fundedAddress,
+			Amount:          big.NewInt(0),
+		},
+		RequiredGasLimit:        big.NewInt(1),
+		AllowedRelayerAddresses: []common.Address{},
+		Message:                 []byte{1, 2, 3, 4},
+	}
+	signedTx := utils.CreateSendCrossChainMessageTransaction(ctx, subnetAInfo, sendCrossChainMessageInput, fundedAddress, fundedKey, teleporterContractAddress)
+
+	log.Info("Sending Teleporter transaction on source chain", "destinationChainID", subnetBInfo.BlockchainID, "txHash", signedTx.Hash())
+	receipt := utils.SendTransactionAndWaitForAcceptance(ctx, subnetAInfo.ChainWSClient, subnetAInfo.ChainRPCClient, signedTx, true)
+
+	event, err := utils.GetSendEventFromLogs(receipt.Logs, subnetATeleporterMessenger)
+	Expect(err).Should(BeNil())
+	Expect(event.DestinationChainID[:]).Should(Equal(subnetBInfo.BlockchainID[:]))
+
+	teleporterMessageID = event.Message.MessageID
+
+	//
+	// Relay the message to the destination
+	//
+
+	receipt = network.RelayMessage(ctx, receipt, subnetAInfo, subnetBInfo, true)
+	teleporterTx, _, err := subnetBInfo.ChainRPCClient.TransactionByHash(ctx, receipt.TxHash)
+
+	//
+	// Check Teleporter message received on the destination
+	//
+	log.Info("Checking the message was received on the destination")
+	delivered, err := subnetBTeleporterMessenger.MessageReceived(&bind.CallOpts{}, subnetAInfo.BlockchainID, teleporterMessageID)
+	Expect(err).Should(BeNil())
+	Expect(delivered).Should(BeTrue())
+
+	//
+	// Attempt to send the same message again
+	//
+	log.Info("Submitting the same Teleporter message again on the destination")
+	gasFeeCap, gasTipCap, nonce := utils.CalculateTxParams(ctx, subnetBInfo.ChainRPCClient, fundedAddress)
+	secondTeleporterTx := types.NewTx(&types.DynamicFeeTx{
+		ChainID:    subnetBInfo.ChainIDInt,
+		Nonce:      nonce,
+		To:         &teleporterContractAddress,
+		Gas:        teleporterTx.Gas(),
+		GasFeeCap:  gasFeeCap,
+		GasTipCap:  gasTipCap,
+		Value:      big.NewInt(0),
+		Data:       teleporterTx.Data(),
+		AccessList: teleporterTx.AccessList(),
+	})
+
+	log.Info("debug",
+		"gas", secondTeleporterTx.Gas(),
+		"gasFeeCap", secondTeleporterTx.GasFeeCap(),
+		"gasTipCap", secondTeleporterTx.GasTipCap(),
+		"data", secondTeleporterTx.Data(),
+		"accesslist", secondTeleporterTx.AccessList(),
+	)
+	signedTx = utils.SignTransaction(secondTeleporterTx, fundedKey, subnetBInfo.ChainIDInt)
+	utils.SendTransactionAndWaitForAcceptance(ctx, subnetBInfo.ChainWSClient, subnetBInfo.ChainRPCClient, signedTx, false)
```

Comment:
is there anything we need to check after transaction sent?

Reply:
The final `bool` parameter of `SendTransactionAndWaitForAcceptance` indicates whether the tx is expected to fail, which it is in this case.